Fixed grammar and spelling issues (#1632)
carbon-language/carbon-lang: docs/project/goals.md

psychological safety. We need consistent expectations for how every community
member should behave, regardless of their position in the community. These
expectations around conduct and behavior need to be clearly articulated both to

> set expectations for people joining and to help remind and anchor us on

Reviewer: ```suggestion
set expectations for people joining, and to help remind and anchor us on
```

I view the comma here as stylistic: there's two `and`s in this sentence so having the comma here can help clarify the intended reading. There might be some better rephrasing, but it's probably better to keep the comma for now.

consistent standards. It is also important that we hold ourselves accountable to
these expectations and have real and meaningful mechanisms to moderate the
community. When behavior steps outside of our expectations, we need tools, 
processes, and policy for how we will recognize and correct it.

**An open, inclusive process for Carbon changes.** The community needs to be
able to effectively engage in the direction and evolution of the project and
language, while keeping the process efficient and effective. That means we need
an open, inclusive process where everyone feels comfortable participating.

> Community members should understand how and why decisions are made and have the

Reviewer: ```suggestion
Community members should understand how and why decisions are made, and have the
```

As above

and other documentation will be provided to make it easy to learn how to use
Carbon.

**Compelling adoption tooling.** We want to provide a compelling suite of tools 
out of the box to encourage the adoption of Carbon at a scale where it can
augment existing C++ codebases. For example, we expect a C++ -> Carbon code
translator will be important.

**Tooling for updating code when Carbon evolves.** As Carbon evolves over time,
we expect to provide tooling to help automate and scale migrating existing
Carbon code to the new version. The goal is to enable a more rapid evolution of
the language without the churn tax and version skew becoming unsustainable.

> **Developer tooling.** We need developers to be productive in reading and writing

Reviewer: ```suggestion
**Developer tooling.** We need developers to be productive while reading and writing
```

I'm fine adding a word here, but probably `while` is more suitable.

**Code should perform predictably.** The reader and writer of code should be
able to easily understand its expected performance, given sufficient background

> knowledge of the environment in which it will run. This need not be precise but 

Reviewer: ```suggestion
knowledge of the environment in which it will run. This need not be precise, but 
```

I think this one is correct.

contract that cannot be strengthened or weakened without breaking the expected
usage patterns is inherently hostile to refactoring. Analogously, features or
conventions that require simultaneously updating all users of an API when
extending it are inherently hostile toward the long-term maintenance of software.

**Support maintaining and evolving the language itself for decades.** We will
not get the design of most language features correct on our first, second, or
73rd try. As a consequence, there must be a built-in plan and ability to move

> Carbon forward at a reasonable pace and at a reasonable cost. Simultaneously,

Reviewer: ```suggestion
Carbon forward at a reasonable pace and with a reasonable cost. Simultaneously,
```

These seem fairly even word choices to me, let's keep the old form unless there's a specific reason for changing.

> an evolving language must not leave software behind to languish but bring

Reviewer: ```suggestion
an evolving language must not leave software behind to languish, instead bringing
```

Comma is definitely correct here. Suggesting a slight phrasing change to emphasize contrast.

misuse.** Adding arbitrary restrictions to prevent misuse of otherwise general
features of the language can create problems when they end up interfering with
unexpected or rare but still appropriate usages. Instead, Carbon should focus on

> enabling appropriate and effective usage of features and creating incentives

Reviewer: ```suggestion
enabling appropriate and effective usage of features, and creating incentives
```

Per other comments -- let's leave commas when there are multiple `and`s in the sentence.

**Design features to be simple to implement.** Syntax, structure, and language
features should be chosen while keeping the implementation complexity

> manageable. The simplicity of implementation reduces bugs, and will in most cases

Reviewer: ```suggestion
manageable. Simplicity of implementation reduces bugs, and will in most cases
```

I'm not sure that adding `The` is improving this sentence.

Iteration requires frequent rebuilds of software as part of the edit/test/debug
cycle of development. The language design should enable low-latency build
strategies, particularly when relatively little has changed. This minimally
requires the separate compilation of source files, and potentially other incremental
build strategies. Separate compilation also enables better scalability options for 

> building systems of large software.

Reviewer: ```suggestion
build systems of large software.
```

This is about build systems.

use a hockey metaphor, we should not skate to where the puck is, much less where
the puck was twenty years ago. We have existing systems to support those
platforms where necessary. Instead, Carbon should be forward-leaning in its

> platform support. As these platforms evolve, Carbon will have to

Reviewer: ```suggestion
platform support. As these platforms evolve over time, Carbon will have to
```

I think there's a certain voice we should retain here.

**Familiarity for experienced C++ developers with a gentle learning curve.** We
need a feasible plan for retraining a C++ workforce to become proficient in

> Carbon. If a long and significant study is required to be minimally proficient,

Reviewer: ```suggestion
Carbon. If long and significant study is required to be minimally proficient,
```

I think it's correct without `a`.

slightly more constraints both in supported features and performance overhead.
In all cases, the particular performance overhead imposed by moving between C++
and Carbon will need to be easily exposed and understood by developers. While a
given piece of code only needs to be migrated once, we expect interoperability to
be invoked continuously to support migrated code and will thus remain important
for most developers.

## Non-goals

There are common or expected goals of many programming languages that we
explicitly call out as non-goals for Carbon. That doesn't make these things bad

> in any way but reflects the fact that they do not provide meaningful value to

Reviewer: ```suggestion
in any way, but instead reflects the fact that they do not provide meaningful value to
```

We can increase contrast, but note:
https://www.grammarly.com/blog/comma-before-but/#:~:text=When%20Do%20You%20Need%20a,have%20to%20trip%20you%20up.

This doesn't preclude having low-level language features or tools to create
specific and curated stable ABIs, or even serializable protocols. Using any such
facilities will also cause developers to explicitly state where they are relying

> on ABI and isolate it in source from code that does not need that stability.

Reviewer: ```suggestion
on ABI and isolating it in source from code which does not need that stability.
```

relying/isolating matches form. Let's keep the original phrasing.
